engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/ModifyProcessInstanceTest.java

```diff
@@ -487,4 +491,300 @@ public void shouldCompletedProcessInstanceWhenElementsInsideExistingFlowScopesAr
         .describedAs("Expect the process instance to have been completed")
         .isPresent();
   }
+
+  @Test
+  public void shouldTerminateElementInRootScope() {
```

Comment:
:wrench: We could extract the termination test cases into a separate class.

The test class gets very long (~800 lines) which makes it harder to maintain. We could split the test cases into activation and termination.